Source/EasyNetQ/Logging/MessageFormatters.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace EasyNetQ.Logging
+{
+    internal static class MessageFormatter
+    {
+        private static readonly Regex Pattern = new(@"(?<!{){@?(?<arg>[^ :{}]+)(?<format>:[^}]+)?}", RegexOptions.Compiled);
+
+        /// <summary>
+        ///     Some logging frameworks support structured logging, such as serilog. This will allow you to add names to structured
+        ///     data in a format string:
+        ///     For example: Log("Log message to {user}", user). This only works with serilog, but as the user of LibLog, you don't
+        ///     know if serilog is actually
+        ///     used. So, this class simulates that. it will replace any text in {curly braces} with an index number.
+        ///     "Log {message} to {user}" would turn into => "Log {0} to {1}". Then the format parameters are handled using regular
+        ///     .net string.Format.
+        /// </summary>
+        /// <param name="messageBuilder">The message builder.</param>
+        /// <param name="formatParameters">The format parameters.</param>
+        /// <returns></returns>
+        public static Func<string> SimulateStructuredLogging(Func<string> messageBuilder, object[] formatParameters)
```

Comment:
It is used only by tests. Also I recommend that you do not bother with all this code. Just use structured logging. It is easier and you do not have to maintain all this code with regexps. Remember that LibLog was deprecated and MS logging framework **supports** structured logging, so... think of removing that file. IMO It's OK to move to structured logging for v7.

Reply:
It is needed to support Log4Net/NLog.

Reply:
That is this method is supposed to be used by other adapter projects not from this repo?

Also NLog supports structured logging as well. I didn't use Log4Net though.

Reply:
Yes. You could look at LibLog implementation for log4net at least.

Reply:
https://github.com/damianh/LibLog/blob/master/src/LibLog/LogProviders/Log4NetLogProvider.cs#L329

Reply:
Worth to add a comment about such design decision.

Reply:
just ditch LibLog :)

Reply:
I mean we need this method to implement integrations with Log4Net and NLog as it was done LibLog.

Reply:
https://github.com/damianh/LibLog/blob/master/src/LibLog/LogProviders/NLogLogProvider.cs#L329

Reply:
>just ditch LibLog :)

I'm trying 🤣

Reply:
Anyway, let me leave `ConsoleLogger` as it is. It was really helpfull for me to set up logging in small apps/examples.

Reply:
> I mean we need this method to implement integrations with Log4Net and NLog as it was done LibLog.

Not we but rather someone else and outside this repo. So it's not our responsibility to maintain such method.